homeassistant/components/switchbot_cloud/switch.py

```diff
@@ -46,22 +46,16 @@ async def async_turn_off(self, **kwargs: Any) -> None:
         self._attr_is_on = False
         self.async_write_ha_state()
 
-    @callback
-    def _handle_coordinator_update(self) -> None:
-        """Handle updated data from the coordinator."""
+    def _set_attributes(self) -> None:
+        """Set attributes from coordinator data."""
         if not self.coordinator.data:
             return
         self._attr_is_on = self.coordinator.data.get("power") == PowerState.ON.value
-        self.async_write_ha_state()
 
 
 class SwitchBotCloudRemoteSwitch(SwitchBotCloudSwitch):
     """Representation of a SwitchBot switch provider by a remote."""
 
-    @callback
-    def _handle_coordinator_update(self) -> None:
```

Comment:
This was previously overriding the method so that it had no actions. Removing it will make the on from the parent class `SwitchBotCloudRemoteSwitch` be used. We should override `set_attributes` here too, right?

Reply:
Indeed, PR updated